File: Android/app/src/main/java/com/example/supplementtracker/presentation/designsystem/OakTypography.kt
package com.example.supplementtracker.presentation.designsystem

import androidx.compose.material3.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

object OakTypography {
    val Display: FontFamily = FontFamily.Serif
}

/**
 * OAK Healthy Typography — wraps OakTypeScale into a full Material3 Typography.
 * All sp values come from OakTypeScale to ensure single source of truth.
 */
val OakTypographySystem = Typography(
    // Display / Hero — Serif, used for screen titles and hero numbers
    displayLarge = TextStyle(
        fontFamily = OakTypography.Display,
        fontWeight = FontWeight.ExtraBold,
        fontSize = OakTypeScale.HeroNumber,       // 42sp
        lineHeight = 48.sp,
        letterSpacing = (-1.5).sp
    ),
    displayMedium = TextStyle(
        fontFamily = OakTypography.Display,
        fontWeight = FontWeight.Bold,
        fontSize = OakTypeScale.ScreenTitle,      // 30sp
        lineHeight = 36.sp,
        letterSpacing = (-0.5).sp
    ),
    displaySmall = TextStyle(
        fontFamily = OakTypography.Display,
        fontWeight = FontWeight.Bold,
        fontSize = OakTypeScale.Metric,           // 24sp
        lineHeight = 30.sp,
        letterSpacing = 0.sp
    ),

    // Headlines — Sans-serif, section-level titles
    headlineLarge = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = OakTypeScale.SectionTitle,     // 20sp
        lineHeight = 26.sp,
        letterSpacing = (-0.3).sp
    ),
    headlineMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 18.sp,
        lineHeight = 24.sp,
        letterSpacing = (-0.2).sp
    ),
    headlineSmall = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 16.sp,
        lineHeight = 22.sp,
        letterSpacing = 0.sp
    ),

    // Body — primary reading text
    bodyLarge = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = OakTypeScale.Body,             // 16sp
        lineHeight = 24.sp,
        letterSpacing = 0.sp
    ),
    bodyMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.sp
    ),
    bodySmall = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = OakTypeScale.Caption,          // 13sp
        lineHeight = 18.sp,
        letterSpacing = 0.sp
    ),

    // Labels — UI controls, chips, badges
    labelLarge = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.1.sp
    ),
    labelMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 12.sp,
        lineHeight = 16.sp,
        letterSpacing = 0.5.sp
    ),
    labelSmall = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 11.sp,
        lineHeight = 14.sp,
        letterSpacing = 0.5.sp
    ),

    // Title — card/list item titles
    titleLarge = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = OakTypeScale.SectionTitle,     // 20sp
        lineHeight = 26.sp,
        letterSpacing = (-0.3).sp
    ),
    titleMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = OakTypeScale.Body,             // 16sp
        lineHeight = 22.sp,
        letterSpacing = 0.sp
    ),
    titleSmall = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.sp
    )
)

File: iOS/Views/OAKTypography.swift
import SwiftUI

/// OAK Healthy Typography System — iOS equivalent of Android OakTypographySystem.
/// All size values come from OAKTypeScale to ensure single source of truth across platforms.
enum OAKFont {
    // MARK: - Display / Hero (Serif)
    /// Hero number — 42pt, used for streak count, big metrics
    static let heroNumber = Font.system(size: OAKTypeScale.heroNumber, weight: .heavy, design: .serif)
    /// Screen title — 30pt, top-level page headings
    static let screenTitle = Font.system(size: OAKTypeScale.screenTitle, weight: .bold, design: .serif)
    /// Metric — 24pt, dashboard numbers
    static let metric = Font.system(size: OAKTypeScale.metric, weight: .bold, design: .serif)

    // MARK: - Headlines (Sans-serif)
    /// Section title — 20pt, card/section headings
    static let sectionTitle = Font.system(size: OAKTypeScale.sectionTitle, weight: .semibold, design: .default)
    /// Headline medium — 18pt
    static let headlineMedium = Font.system(size: 18, weight: .semibold, design: .default)
    /// Headline small — 16pt
    static let headlineSmall = Font.system(size: 16, weight: .semibold, design: .default)

    // MARK: - Body
    /// Body large — 16pt, primary reading text
    static let bodyLarge = Font.system(size: OAKTypeScale.body, weight: .regular, design: .default)
    /// Body medium — 14pt
    static let bodyMedium = Font.system(size: 14, weight: .regular, design: .default)
    /// Body small / Caption — 13pt, secondary info
    static let caption = Font.system(size: OAKTypeScale.caption, weight: .regular, design: .default)

    // MARK: - Labels / UI Controls
    /// Label large — 14pt semibold, buttons, chips
    static let labelLarge = Font.system(size: 14, weight: .semibold, design: .default)
    /// Label medium — 12pt medium, badges, tags
    static let labelMedium = Font.system(size: 12, weight: .medium, design: .default)
    /// Label small — 11pt medium, timestamps, fine print
    static let labelSmall = Font.system(size: 11, weight: .medium, design: .default)
}

// MARK: - View Modifier convenience
extension View {
    func oakFont(_ font: Font) -> some View {
        self.font(font)
    }
}
